gsadjust/pyqt_models.py

```diff
@@ -509,34 +510,49 @@ def from_json(cls, simple_survey):
                 # Raised if delta type is 'assigned'
                 pass
             deltas.append(d)
+
         temp.deltas = deltas
-        for datum in simple_survey['datums']:
+       
+        for datum in data['datums']:
             d = data_objects.Datum(datum['station'])
             d.__dict__ = datum
             temp.datum_model.insertRows(d, 0)
 
         ao = data_objects.AdjustmentOptions()
-        ao.__dict__ = simple_survey['adjoptions']
+        ao.__dict__ = data['adjoptions']
         temp.adjustment.adjustmentoptions = ao
-        # temp.adjustment.adjustmentoptions = simple_survey.adjoptions
 
-        if 'checked' in simple_survey:
-            temp.setCheckState(simple_survey['checked'])
+        if 'checked' in data:
+            temp.setCheckState(data['checked'])
         return temp
 
-    @classmethod
-    def from_simplesurvey(cls, simple_survey):
-        """
-        When loading a workspace, repopulate PyQt models
-        """
-        temp = cls(simple_survey.name)
-        temp.deltas = simple_survey.deltas
-        for datum in simple_survey.datums:
-            temp.datum_model.insertRows(datum, 0)
-        temp.adjustment.adjustmentoptions = simple_survey.adjoptions
-        if hasattr(simple_survey,'checked'):
-            temp.setCheckState(simple_survey.checked)
-        return temp
+    def to_json(self):
+        loops = []
+        deltas = []
+        datums = []
+        # Remove ObsTreeStation objects from deltas in the survey delta_model (which is different than the individual
+        # loop delta_models; those are recreated when the workspace is loaded.
+        for i in range(self.delta_model.rowCount()):
+            ind = self.delta_model.createIndex(i, 0)
+            delta = self.delta_model.data(ind, QtCore.Qt.UserRole)
+            simpledelta = data_objects.SimpleDelta(delta)
+            deltas.append(simpledelta)
+        for i in range(self.datum_model.rowCount()):
+            ind = self.datum_model.createIndex(i, 0)
+            datums.append(self.datum_model.data(ind, QtCore.Qt.UserRole))
+        for i in range(self.rowCount()):
+            obstreeloop = self.child(i)
+            simpleloop = data_objects.SimpleLoop(obstreeloop)
+            loops.append(simpleloop)
+
+        return {
+            'loops': jsons.dump(loops),
```

Comment:
We need to use jsons.dump for the children still, until their .to_json methods are in place -- jsons doesn't look deeper than what you return in a custom serializer, so json (Python module) would error out on the Simple objects.